### PR TITLE
Accessibility Fixes

### PR DESCRIPTION
**Pull request by anotherminh**
## Related Issues

- https://transcend.height.app/T-28192

https://github.com/transcend-io/consent-manager-ui/assets/75641375/519ebfed-8ca3-4a81-8333-49fade91f017

### CODE REVIEW

**Comment by blake-transcend** [comment]
@anotherminh I pushed the focus fix to this PR

**Review by blake-transcend — DISMISSED** [review]

File: src/components/Main.tsx [hunk 1]
```diff
@@ -60,14 +61,33 @@ export function Main({
...
-        <div role="document" className="modal-container-inner">
+      <div
+        role="dialog"
+        aria-model="true"
+        aria-labelledby="consent-dialog-title"
+        className="modal-container"
+        id="consentManagerMainDialog"
+        ref={
+          dialogRef.current
+            ? (dialogRef as unknown as Ref<HTMLDivElement>)
```
Comment (anotherminh) [inline src/components/Main.tsx:86]: cc @blake-transcend i had to do all this to get tsc to be happy...

Comment (blake-transcend) [reply src/components/Main.tsx:86]: wtf we should always be setting this ref, especially when the current ref is undefined

File: src/components/Main.tsx [hunk 2]
```diff
@@ -60,14 +61,29 @@ export function Main({
   /** Set of supported languages */
   supportedLanguages: ConsentManagerLanguageKey[];
 }): JSX.Element {
+  // need to focus the first button in the modal when the modal is opened
+  const dialogRef = useRef<HTMLDivElement>(null);
```
Comment (blake-transcend) [inline src/components/Main.tsx:65]: when I changed this to have a null initial value it worked

**Review by eligrey — APPROVED** [review]
I'm curious about the ID attributes. Is there any reason why this can't/shouldn't be a class?

Otherwise this lgtm!

**Comment by anotherminh** [comment]
@eligrey That's what MDN recommends https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role (and also the customer asked for this) 

If there's some other doc I should reference tho, feel free to share!

**Comment by eligrey** [comment]
oh that makes sense to me

**Pull request merged**